JD-4-5-6/src/test/java/com/example/weekone/WeekOneApplicationTests.java

```diff
@@ -27,5 +27,13 @@ void reverseFibonacciSequence(){
         assertEquals(expectedReverseOutput, actualReverseOutput);
     }
 
+    @Test
+    void insertionSort(){
+        int lengthOfSequence = 11;
+        List<Integer> fibonacciSequence = WeekOneApplication.generateFibonacciSequence(lengthOfSequence);
+        ArrayList<Integer> expectedOut = new ArrayList<Integer>(Arrays.asList(0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 233));
+        assertEquals(expectedOut, WeekOneApplication.insertionSort(fibonacciSequence));
+    }
```

Comment:
The happy path test is a great way to validate your assumptions, but it is often inadequate for verifying edge cases. Can you add another test which tests a random assortment of numbers? For kicks lets do 
`
List<Integer> randomList = new ArrayList<Integer>(Arrays.asList(233,0,43,3,1,24,31,18,4,4,5,-1,-233))
ArrayList<Integer> expectedOut = new ArrayList<Integer>(Arrays.asList(-233,-1,0,1,3,4,4,5,18,24,31,43,233));
`

Reply:
Thanks for that feedback! I put that in, and found the test passed with one change. I removed the set of 233 at index of 0. Now the insertionSort can be reused with any Collection List for the input size.